docs/development/pigs/pig-024.rst

```diff
@@ -0,0 +1,267 @@
+In this case, the citation scheme of Gammapy should be used by these authors.
+
+As the length of the author list is generally a constrain, the author list is reduced to the short list of contributors
+for the conference, followed by 'The Gammapy team' associated with a link to the Gammapy team webpage:
+
+ ::
+
+  oo, ff, tt, for `the Gammapy team <https://gammapy.org/team.html>`_
+
+If there is a corresponding author, the 'Gammapy Coordination Committee' associated with its usual mailing list, is
+used. Concerning the acknowledgement, the Astropy project should always be mentioned, and if possible our mandatory
+external libraries.
+
+
+4. Metadata files
+==================================
+
+Depending on the software repository, different metadata files are used in the eco-system of Open Source research
+software.
+
+The file ``codemeta.json`` is used by HAL and Software Heritage, and recommended by
+`ESCAPE <https://escape2020.pages.in2p3.fr/wp3/eossr/v0.3.2/metadata.html>`_ or
+`EOSC <https://www.eosc-pillar.eu/news/illustrating-benefits-eosc-research-communities-germany>`_.
+
+The file ``CITATION.cff`` is used by Zenodo and GitHub. Its format should follow the rules set up by the
+`Citation File Format project <https://pythonlang.dev/repo/citation-file-format-citation-file-format/>`_. A GitHub
+Action can be used to automatically check the compliance with the latest format. At the date of this PIG, there is no
+official scheme to handle current and past affiliations, that might be needed for LTS releases for example (see the
+CFF project `Issue #268 <https://github.com/citation-file-format/citation-file-format/issues/268>`_). In the affiliation
+string, one could add a past affiliation as precised below, but with a risk that Zenodo or tools using ``CITATION.cff``
+to make the ``codemeta.json`` does not understand it.
+
+ ::
+
+    authors:
+      - family-names: XXX
+        given-names: "YYYY Z."
+        affiliation: "Lab AA, XX; Past: Lab BB, YY"
+        email: yyyy.xxx@gammapy.org
+        orcid: "https://orcid.org/0000-0000-0000-0000"
+
+For any release, both metadata files should be carefully updated and systematically a review should be done by the dev
+team.
+
+5. Possible implementation
+==========================
+
+In order to respect these rules, some automation is required to create the list of contributors. One could use
+the Python library `Tributors <https://con.github.io/tributors/>`_. This library uses the GitHub API to determine the
+list of contributors, which is written in the format associated with codemeta.json, citation.cff, zenodo.json. This kind
+of library allows then two requirements: retrieve automatically the list of contributors from GitHub and write the
+authors list in all the needed formats.
+
+One should note that the user profile of the GitHub platform does not contain a field associated with the ORCID
+identifier. However, one could also retrieve the ORCID identifier of a contributor with the ORCID API
```

Comment:
I think we will just ask for there ORCID from contributors as we do for the affiliation. The ORCID typically does not change and it will be stored and maintained in the `CITATION.cff`...

Reply:
There is a small subtlety: the citation file contains the list of contributors since the latest feature release, not from all the beginning of the project. So the ORCID id can not be stored into this file for all users.

Reply:
Thanks for clarifying, indeed this I have overseen! This raises the question for me whether we maybe introduce an independent `AUTHORS.yaml` for maintenance which contains the author entries independent of the current release. This could be used for LTS releases as well. What do you think?

Reply:
This could technically help and help the maintenance. To avoid the confusion with the CITATION.cff for external person, one could call it `LTS_AUTHORS.yaml`. 
I am pro this concept... and if you are with the name, I go for it both in this PR and in the PR that I prepare about the script `dev/authors.py`

Reply:
Sounds good to me. I think having an author list "archive" with information independent of any version will help maintenance, because we can create the "running" author list in `CITATION.cff` from it.

Reply:
I find it unusual to drop people from the citation. In my opinion, the CITATION.cff should include all people who have contributed to Gammapy by the definitions above, not just the ones contributing between the last and the current release.


In the extreme example, a minor version release with a single bugfix would have a single author. That's clearly not wanted, is it?


That also makes it possible to use this file as the one source of truth about the contributors where they can provide there own data themselves via pull requests, which solves the consent problem.

For past contributors, we could write them to give there consent explicitly in the first update to the citation.cff we do before the next release.

Reply:
They should be a mis-understanding as nobody is dropped, unless under her/his request.
Otherwise, I add a sentence to be clear on the bug release.

Reply:
I think whether dropping people from author lists is fine or not depends on the time scale. E.g. it is very common for large collaborations that people are removed from the author list once they have left the collaboration since a while. To my knowledge the policy proposed in the PIG is:
```
The list of authors is composed of the union of all the contributors of the releases realised since the last LTS release.
```
This means the `CITATION.cff` is a "running" author list which will consist of all people that contributed since the last LTS release. This  defines a timeline of updating the author list of ~2 years, which I think is fair. The case that you just mentioned with a "single author" cannot happen. On the other hand "old contributors" that have left the project before v1.0 will not be part of the v2.0 author list.

The "all time" contributor list will then be an independently manually maintained `ALL_AUTHORS_EVER.yaml` file. From which the `CITATION.cff` (which is e.g. used for Zenodo) is created automatically. I hope this makes sense.

Reply:
Addendum: v1.0 will include all contributors since the beginning of the project.

Reply:
Ok, I know realize that indeed bug fix releases need to be specified more precisely. Taking the policy very strictly would indeed mean that `v1.0.1` would only include the people that have contributed to the bug-fix release, which does not seem fair. I guess the better default for bug fix release would be the authors from the release the bug-fix relates too plus additional new authors that e.g. have contributed the bug fix.

Reply:
Indeed, I have just changed it after one comment of Max in this way ;)